resources/assets/lib/chat/conversation-view.tsx

```diff
@@ -103,18 +146,29 @@ export default class ConversationView extends React.Component<any, any> {
     const oldPMLink = `https://osu.ppy.sh/forum/ucp.php?i=pm&mode=compose&u=${channel.pmTarget}`;
     const conversationStack: JSX.Element[] = [];
     let currentGroup: Message[] = [];
-    let lastReadIndicatorShown = false;
+    let unreadMarkerShown = false;
     let currentDay: number;
 
     _.each(channel.messages, (message: Message, key: number) => {
       // check if the last read indicator needs to be shown
-      if (!lastReadIndicatorShown && message.messageId > dataStore.uiState.chat.lastReadId && message.sender.id !== currentUser.id) {
-        lastReadIndicatorShown = true;
+      if (!unreadMarkerShown && message.messageId > dataStore.uiState.chat.lastReadId && message.sender.id !== currentUser.id) {
+        unreadMarkerShown = true;
+
+        // If the unread marker is the first element in this conversation, it most likely means that the unread cursor
+        // is even further in the past, making the displayed marker somewhat useless (until we can back-load those
+        // past messages in)... thus we ignore it when auto-scrolling and just go to the bottom instead.
+        //
+        // TODO: Actually in hindsight, there's another scenario where the first element in the conversation is an
+        // unread marker - when you receive new PMs and have yet to read any. Will look to handle this case later...
+        if (!_.isEmpty(conversationStack)) {
```

Comment:
looking closer, this check is wrong? It's when the stack is empty then there might be backlog?

```suggestion
        if (_.isEmpty(conversationStack)) {
```